wasm/jit/jit_amd64_test.go

```diff
@@ -779,6 +833,96 @@ func TestAmd64Builder_handleConstI64(t *testing.T) {
 	}
 }
 
+func TestAmd64Builder_handleConstF32(t *testing.T) {
+	for _, v := range []float32{1, -3.23, 100.123} {
+		t.Run(fmt.Sprintf("%f", v), func(t *testing.T) {
+			o := &wazeroir.OperationConstF32{Value: v}
+			builder := requireNewBuilder(t)
+			builder.initializeReservedRegisters()
+			// Dummy value not used!
+			_ = builder.locationStack.pushValueOnStack()
+
+			// Now emit the const instruction.
+			err := builder.handleConstF32(o)
+			require.NoError(t, err)
+
+			// To verify the behavior, we double and push the const value
+			// to the stack.
+			loc := builder.locationStack.peek()
+			require.Equal(t, wazeroir.SignLessTypeF32, loc.valueType)
+			prog := builder.newProg()
+			prog.As = x86.AADDSS
+			prog.To.Type = obj.TYPE_REG
+			prog.To.Reg = loc.register
+			prog.From.Type = obj.TYPE_REG
+			prog.From.Reg = loc.register
+			builder.addInstruction(prog)
+			builder.releaseRegisterToStack(loc)
+			builder.returnFunction()
+
+			// Assemble.
+			code, err := builder.assemble()
+			require.NoError(t, err)
+			// Run code.
+			eng := newEngine()
+			mem := newMemoryInst()
+			jitcall(
+				uintptr(unsafe.Pointer(&code[0])),
```

Comment:
trailing comments for each of these lines as they look magical

Reply:
https://github.com/tetratelabs/wazero/pull/82/commits/16ab8444ef131018ae1377a3b9452df98e9fa7bc commented the original code!